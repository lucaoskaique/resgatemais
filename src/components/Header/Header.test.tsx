import { render, screen } from "@testing-library/react"
import { describe, expect, it, vi } from "vitest"

import { Header } from "."

const mockScrollTo = vi.fn()
global.scrollTo = mockScrollTo

describe("<Header />", () => {
  it("should render the heading text", () => {
    render(<Header />)
    expect(
      screen.getByText("Organização Audiovisual da Sociedade Civil")
    ).toBeDefined()
  })

  it('should render the correct images with alt text "Logo"', () => {
    render(<Header />)
    const images = screen.getAllByAltText("Logo")
    expect(images).toHaveLength(3)
    expect(images[0]).toBeDefined()
    expect(images[1]).toBeDefined()
  })

  it("should apply the correct classes based on scroll position", () => {
    render(<Header />)

    window.scrollY = 100
    window.dispatchEvent(new Event("scroll"))
    const header = screen.getByTestId("header")

    expect(header).toContain("bg-white")
    expect(header).not.toContain("bg-transparent")
  })

  it("should toggle the background color when scrolled", () => {
    render(<Header />)

    window.scrollY = 0
    window.dispatchEvent(new Event("scroll"))
    let header = screen.getByTestId("header")
    expect(header).toContain("bg-transparent")
    expect(header).not.toContain("bg-white")

    window.scrollY = 100
    window.dispatchEvent(new Event("scroll"))
    header = screen.getByTestId("header")
    expect(header).toContain("bg-white")
    expect(header).not.toContain("bg-transparent")
  })
})
