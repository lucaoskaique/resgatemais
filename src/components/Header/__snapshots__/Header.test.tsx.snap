// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > should match the snapshot 1`] = `
<header
  class="fixed w-full py-6 px-10 top-0"
  id="header"
>
  <nav
    class="flex items-center justify-between w-full text-xs text-black dark:text-white font-bold"
  >
    <div
      class="flex items-center gap-6"
    >
      <div
        class="hidden dark:block"
      >
        <a
          href="#"
        >
          <img
            alt="Logo"
            class="w-full max-w-[6.5rem]"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/logo-oficial-02.svg"
            style="color: transparent;"
            width="100"
          />
        </a>
      </div>
      <div
        class="block dark:hidden"
      >
        <a
          href="#"
        >
          <img
            alt="Logo"
            class="w-full max-w-[6.5rem]"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/logo-oficial_Prancheta-1.svg"
            style="color: transparent;"
            width="100"
          />
        </a>
      </div>
      <p>
        Organização Audiovisual da Sociedade Civil
      </p>
    </div>
    <ul
      class="flex itens-center gap-9"
    >
      <li
        class="py-3"
      >
        <a
          href="#"
        >
          Sobre
        </a>
      </li>
      <li
        class="py-3"
      >
        <a
          href="#"
        >
          Ações
        </a>
      </li>
      <li>
        <button>
          <span
            class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                aria-label="Switch demo"
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
        </button>
      </li>
      <li
        class="py-3"
      >
        <a
          class="bg-color-red px-4 py-3 rounded-[0.313rem]"
          href="#"
        >
          Doar
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`<Header /> > should render the heading 1`] = `
<header
  class="fixed py-6 px-10 top-0 flex items-center justify-between w-full text-xs text-black dark:text-white font-bold"
  id="header"
>
  <div
    class="flex items-center gap-6"
  >
    <div
      class="hidden dark:block"
    >
      <a
        href="#"
      >
        <img
          alt="Logo"
          class="w-full max-w-[6.5rem]"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="/logo-oficial-02.svg"
          style="color: transparent;"
          width="100"
        />
      </a>
    </div>
    <div
      class="block dark:hidden"
    >
      <a
        href="#"
      >
        <img
          alt="Logo"
          class="w-full max-w-[6.5rem]"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="/logo-oficial_Prancheta-1.svg"
          style="color: transparent;"
          width="100"
        />
      </a>
    </div>
    <h1>
      Organização Audiovisual da Sociedade Civil
    </h1>
  </div>
  <nav>
    <ul
      class="flex itens-center gap-9"
    >
      <li
        class="py-3"
      >
        <a
          href="#"
        >
          Sobre
        </a>
      </li>
      <li
        class="py-3"
      >
        <a
          href="#"
        >
          Ações
        </a>
      </li>
      <li>
        <button>
          <span
            class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                aria-label="Switch demo"
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
        </button>
      </li>
      <li
        class="py-3"
      >
        <a
          class="bg-color-red px-4 py-3 rounded-[0.313rem]"
          href="#"
        >
          Doar
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
